src/communication/makeRetrying.ts

```diff
@@ -74,8 +83,13 @@ function parseRetryAfterHeader(response: AxiosResponse): number | undefined {
  * safely re-attempt requests.
  */
 export default function makeRetrying(axiosInstance: AxiosInstance) {
-  // Set the attempt (in the request configuration) of any request.
+  // Intercept all outgoing requests.
   axiosInstance.interceptors.request.use((config: AxiosRequestConfig & Partial<AttemptState>) => {
+    // If the request is a POST or DELETE one and does not yet have the idempotency header, add one now.
+    if (unsafeMethods.has(config.method!) && config.headers?.[idempotencyHeaderName] == undefined) {
+      config.headers = { ...config.headers, ...generateIdempotencyHeader() };
```

Comment:
```suggestion
      Object.assign(config.headers, generateIdempotencyHeader());
```

Reply:
Axios merges the headers passed into `axios.request` with the ones passed to `axios.create`, so the the object passed to the interceptor _should_ be a fresh copy anyway and therefore be safe to mutate.

Is _should_ certain and safe enough for us?

Reply:
I agree that it reads better. And I might be worrying too much, so let's go for it.

Reply:
Turns out this doesn't work after all, as `headers` could be `undefined`. 🙄 
I'll fix this in my next pr.

Reply:
(We've changed this to `Object.assign(config.headers ??= {}, generateIdempotencyHeader())` in [the next PR](https://github.com/mollie/mollie-api-node/pull/298).)